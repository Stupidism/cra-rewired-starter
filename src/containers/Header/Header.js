import React from 'react';
import { Layout } from 'antd';

import connect from 'store/connect';
import logoSpinning from 'store/selectors/logoSpinning';

import logo from 'assets/logo.svg';

import './Header.less';

export const HeaderRenderer = ({ logoSpinning }) => (
  <Layout.Header className="Header">
    <img
      src={logo}
      className={`Logo ${logoSpinning ? '' : 'LogoSpinPaused'}`}
      alt="logo"
    />
    <h1 className="Title">Welcome to React</h1>
  </Layout.Header>
);

export default connect({ logoSpinning })(HeaderRenderer);
