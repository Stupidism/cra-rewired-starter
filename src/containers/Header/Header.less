.Header {
  background-color: #222;
  padding: 20px;
  height: auto;

  .Logo {
    animation: logo-spin infinite 20s linear;
    height: 80px;

    &.LogoSpinPaused {
      animation-play-state: paused;
    }
  }

  .Title {
    font-size: 28px;
    line-height: 28px;
    color: white;
    margin-top: 20px;
    margin-bottom: 0;
  }

  @keyframes logo-spin {
    from { transform: rotate(0deg); }
    to { transform: rotate(360deg); }
  }
}
