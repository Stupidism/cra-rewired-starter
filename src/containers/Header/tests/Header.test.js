import React from 'react';
import { mount } from 'enzyme';

import { HeaderRenderer } from '../Header';

describe('<App />', () => {
  describe('should be consistent with snapshots', () => {
    it('when default', () => {
      const snapshot = mount(<HeaderRenderer />);
      expect(snapshot).toMatchSnapshot();
    });

    it('when logo is spinning', () => {
      const snapshot = mount(<HeaderRenderer logoSpinning />);
      expect(snapshot).toMatchSnapshot();
    });
  });

  it('should render the logo and the title', () => {
    const wrapper = mount(<HeaderRenderer to="some-url" />);
    expect(wrapper.find('img.Logo').length).toBe(1);
    expect(wrapper.find('h1.Title').length).toBe(1);
  });
});
