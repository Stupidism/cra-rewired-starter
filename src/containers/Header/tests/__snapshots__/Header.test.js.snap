// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should be consistent with snapshots when default 1`] = `
<div
  className="Header ant-layout-header"
>
  <img
    alt="logo"
    className="Logo LogoSpinPaused"
    src="logo.svg"
  />
  <h1
    className="Title"
  >
    Welcome to React
  </h1>
</div>
`;

exports[`<App /> should be consistent with snapshots when logo is spinning 1`] = `
<div
  className="Header ant-layout-header"
>
  <img
    alt="logo"
    className="Logo "
    src="logo.svg"
  />
  <h1
    className="Title"
  >
    Welcome to React
  </h1>
</div>
`;
